front/src/common/Map/Search/MapSearchLine.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react';
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+import { useTranslation } from 'react-i18next';
+import { useDebounce } from 'utils/helpers';
+import { useSelector, useDispatch } from 'react-redux';
+import { getInfraID } from 'reducers/osrdconf/selectors';
+import { Viewport, updateLineSearchCode, updateMapSearchMarker } from 'reducers/map';
+import nextId from 'react-id-generator';
+import { BBox } from '@turf/helpers';
+import bbox from '@turf/bbox';
+import WebMercatorViewport from 'viewport-mercator-project';
+import { getMap } from 'reducers/map/selectors';
+import { Zone, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { searchPayloadType } from '../const';
+import LineCard from './LineCard';
+import { ILineSearchResult } from './searchTypes';
+
+type MapSearchLineProps = {
+  updateExtViewport: (viewport: Partial<Viewport>) => void;
+};
+
+const MapSearchLine: React.FC<MapSearchLineProps> = ({ updateExtViewport }) => {
+  const infraID = useSelector(getInfraID);
+  const map = useSelector(getMap);
+  const { t } = useTranslation(['map-search']);
+  const dispatch = useDispatch();
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
+  const [searchState, setSearchState] = useState<string>('');
+  const [dontSearch, setDontSearch] = useState(false);
+  const [searchResults, setSearchResults] = useState<ILineSearchResult[] | undefined>(undefined);
+  const [dataTrackZone, setDataTrackZone] = useState<{ id: number; lineCode: number }>({
+    id: 0,
+    lineCode: 0,
+  });
+  const [skip, setSkip] = useState<boolean>(true);
+
+  const { data: trackZones } = osrdEditoastApi.useGetInfraByIdLinesAndLineCodeBboxQuery(
+    dataTrackZone,
+    { skip }
+  );
+  const zoomToFeature = (boundingBox: BBox) => {
+    const [minLng, minLat, maxLng, maxLat] = boundingBox;
+    const viewportTemp = new WebMercatorViewport({ ...map.viewport, width: 600, height: 400 });
+    const { longitude, latitude, zoom } = viewportTemp.fitBounds(
+      [
+        [minLng, minLat],
+        [maxLng, maxLat],
+      ],
+      { padding: 40 }
+    );
+
+    const tempMap = { ...map };
+    const newViewport = {
+      ...tempMap.viewport,
+      longitude,
+      latitude,
+      zoom,
+    };
+    updateExtViewport(newViewport);
+  };
+
+  const debouncedSearchTerm = useDebounce(searchState, 300);
+
+  const updateSearch = async (params: searchPayloadType) => {
+    try {
+      const results = await postSearch({
+        body: { object: params?.object, query: params?.query },
```

Comment:
```suggestion
        body: { object: params.object, query: params.query },
```
No need for ? as you typed the params

Reply:
Thank you, it was an oversight.